react-app/src/components/property/PropertyDialog.tsx

```diff
@@ -262,44 +263,149 @@ export const PropertyAssessedValueEditDialog = (props: IPropertyAssessedValueEdi
         Id: building.Id,
         Evaluations: building.Evaluations?.map((evalu) => ({
           ...evalu,
-          Value: evalu.Value.replace(/[$,]/g, ''), // Obviously this double map is pretty evil so suggestions welcome.
+          Value: String(evalu.Value).replace(/[$,]/g, ''), // Obviously this double map is pretty evil so suggestions welcome.
         })),
       })),
     });
   }, [initialValues, initialRelatedBuildings]);
 
+  const currentYear = new Date().getFullYear();
+  const years = [currentYear];
+  const defaultParcelValues = years.map((year) => ({
+    Year: year,
+    Value: 0,
+    EffectiveDate: Date(),
+    EvaluationKeyId: 0,
+    ParcelId: initialValues?.Id || null,
+  }));
```

Comment:
Noticing that this is the only time that I think currentYear or years is used. So years will always be an array of 1. 
It seems like in that case we could just shorten this down to this:
```suggestion
  const defaultParcelValues = [{
    Year: new Date().getFullYear(),
    Value: 0,
    EffectiveDate: Date(),
    EvaluationKeyId: 0,
    ParcelId: initialValues?.Id || null,
  }];
```
Also, what if it's a building instead of a parcel?